extensions/zacoons/webgpu.js

```diff
@@ -0,0 +1,638 @@
+                defaultValue: "",
+              },
+              DST: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: "",
+              },
+            },
+          },
+          {
+            opcode: "createCommandBuffer",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "🗍 create command buffer",
+            disableMonitor: true,
+          },
+
+          label("Device Queues"),
+          {
+            opcode: "newQueue",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "≫ new queue",
+          },
+          {
+            opcode: "writeListToBuffer",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "≫ write list [LIST_ID] with type [TYPE] to buffer [BUFFER]",
+            arguments: {
+              LIST_ID: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "lists",
+              },
+              TYPE: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "types",
+              },
+              BUFFER: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: "",
+              },
+            },
+          },
+          {
+            opcode: "addCommandBufferToQueue",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "≫ add command buffer [COMMAND_BUFFER] to queue",
+            arguments: {
+              COMMAND_BUFFER: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: "",
+              },
+            },
+          },
+          {
+            opcode: "submitQueue",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "≫ submit queue",
+          },
+        ],
+        menus: {
+          bindGroupLayoutEntryTypes: {
+            acceptReporters: true,
+            items: ["uniform", "read-only-storage", "storage"],
+          },
+          bufferUsages: {
+            acceptReporters: true,
+            items: [
+              "COPY_SRC",
+              "COPY_DST",
+              "INDEX",
+              "INDIRECT",
+              "MAP_READ",
+              "MAP_WRITE",
+              "QUERY_RESOLVE",
+              "STORAGE",
+              "UNIFORM",
+              "VERTEX",
+            ],
+          },
+          modules: {
+            acceptReporters: true,
+            items: "_getModuleNames",
+          },
+          lists: {
+            acceptReporters: true,
+            items: "_getLists",
+          },
+          types: {
+            acceptReporters: true,
+            items: Object.keys(this.types),
+          },
+        },
+      };
+    }
+
+    init() {
+      this._resetAttrs();
+
+      if (!navigator.gpu) return false;
+
+      return navigator.gpu.requestAdapter().then(async (adapter) => {
+        this.dev = await adapter.requestDevice();
+        this._createModules();
+        return true;
+      });
+    }

```

Comment:
When using any modern browser which technically supports WebGPU, but doesn't support it in the specific conditions which it is being used, navigator.gpu will exist, but the promise returned from navigator.gpu.requestAdapter() will get rejected. This makes the block return undefined and show some warning in the console.
For example, Firefox on Linux shows this:
![image](https://github.com/TurboWarp/extensions/assets/90340988/1b247f56-25e8-4e37-a6fa-9b3811934687)

Reply:
I think I've resolved this, but I can't really test it so let me know.
Thanks for your thorough review, I really appreciate it.